spyder/plugins/editor/plugin.py

```diff
@@ -2950,3 +3002,72 @@ def set_create_new_file_if_empty(self, value):
         """Change the value of create_new_file_if_empty"""
         for editorstack in self.editorstacks:
             editorstack.create_new_file_if_empty = value
+
+
+class AutosaveComponent(object):

```

Comment:
Let's move this class to `plugins/editor/utils/autosave.py` because

a) It's pure Python.
b) We don't need to add more content to the (already very big) Editor plugin file.